checks/cloudwatch.go

```diff
@@ -31,7 +31,11 @@ func (c *CloudWatchChecker) Type() string {
 }
 
 func (c *CloudWatchChecker) Check(ctx *context.Context, extConfig external.Check) *pkg.CheckResult {
-	check := extConfig.(v1.CloudWatchCheck)
+	updated, err := Contextualise(extConfig, ctx)
+	if err != nil {
+		return pkg.Fail(extConfig, ctx.Canary)
+	}
+	check := updated.(v1.CloudWatchCheck)
```

Comment:
does the contextualization not need to happen on check with type information ?

Reply:
Not sure I follow. The need is to generically template any chained canary, so the function is written to take an external.Check rather than a specifically typed v1.xxCheck